src/Rule/Email.php

```diff
@@ -126,4 +126,28 @@ public function enableIDN(bool $enableIDN): self
         $new->enableIDN = $enableIDN;
         return $new;
     }
+
+    /**
+     * @inheritDoc
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'email';
+    }
+
+    /**
+     * @inheritDoc
+     * @return array
+     */
+    public function getOptions(): array
+    {
+        return array_merge(
+            $this->allowName ? ['allowName' => true] : [],
```

Comment:
Why if the bool value does not always output it, but return an empty array?

eg `['checkDNS' => $this->checkDNS]`

Reply:
If the default value for a flag has not been changed and this flag has not been changed, the result is empty.
If it is formed for the default values, the returned array will be too large.
We do not force the user to specify all values when creating a validator.
;-)

Reply:
It's be corrected